config/routes.rb

```diff
@@ -6,9 +6,7 @@
   resources :activities
   resources :newsletters
 
-  match '/cards/update' => 'cards#update', :via => [:put]
-  match '/cards/:user_slug/edit' => 'cards#edit', :via => [:get]
-  match '/subscriptions/upgrade' => 'subscriptions#upgrade', :via => [:put]
```

Comment:
it's subscriptions/update - it's folded into resources :subscriptions - properly RESTful - well maybe - what the user is doing is more like account or user update - they are changing something about the nature of their relation with AV ...